spec/unit/mutant/repository/diff_spec.rb

```diff
@@ -117,5 +129,45 @@ def apply
         end
       end
     end
+
+    context 'work dir is subdirectory of git root' do
```

Comment:
this example group proves "nothing" as `/foo` could only have entered the system via `--show-toplevel` in the first place. Hence the semantics of this tests are nothing extra to the existing one. I did not make an equivalent in #1142

Reply:
the existing ones remained untouched with regard to paths and did not justify adding the ceremony of relative

Reply:
I've got to disagree.

The specs proof that all paths are constructed relative to `--show-toplevel` when we changed out the expectation from `Pathname.pwd` to `git rev-parse --show-toplevel`.

Basically this spec used to describe the situation "relative to `Pathname.pwd`" and once changed (in #1142) to use `git rev-parse --show-toplevel` (and not anymore `Pathname.pwd), the specs now describe the situation relative to the new base. We also "proof" (by example only) that `/foo` could only have entered the system via the one method now used.

We changed the "method the base" enters the code base. The spec you added "did not prove anything extra" on top of what was already proven. As additional evidence: I had not to add your example over in #1142 to cover all mutations.

IMO tests should not "re-specify" that the boundaries (before `Pathname.pwd`, now `git rev-parse --show-toplevel`) do, insted they are supposed to specify what our code does on top. We changed the code to use a new base, and hence I only expect this change to be reflected in the specs.